contribs/gmf/src/directives/print.js

```diff
@@ -365,84 +377,95 @@ gmf.PrintController = function($rootScope, $scope, $timeout, $q, $injector,
   this.hiddenAttributeNames;
 
   /**
-   * @return {ol.Size} Size in dots of the map to print.
+   * @type {function(ol.render.Event)}
    */
-  const getSizeFn = () => this.paperSize_;
+  this.postcomposeListener_;
 
   /**
-   * @param {olx.FrameState} frameState Frame state.
-   * @return {number} Scale of the map to print.
+   * @type {angular.$http.HttpPromise}
+   * @private
    */
-  const getScaleFn = (frameState) => {
-    // Don't compute an optimal scale if the user manualy choose a value not in
-    // the pre-defined scales. (`scaleInput` in `gmfPrintOptions`).
-    goog.asserts.assert(this.layoutInfo.scales);
-    goog.asserts.assert(this.layoutInfo.scale !== undefined);
-    if (this.layoutInfo.scale === -1 ||
-        ol.array.includes(this.layoutInfo.scales, this.layoutInfo.scale)) {
-      const mapSize = frameState.size;
-      const viewResolution = frameState.viewState.resolution;
-      this.layoutInfo.scale = this.getOptimalScale_(mapSize, viewResolution);
-    }
-    return this.layoutInfo.scale;
-  };
-
-  let getRotationFn;
-  if (this.rotateMask_) {
-    /**
-     * @return {number} rotation to apply.
-     */
-    getRotationFn = () => this.rotation;
-  }
+  this.capabilities_;
 
   /**
-   * @type {function(ol.render.Event)}
+   * @type {gmfThemes.GmfOgcServers}
+   * @private
    */
-  this.postcomposeListener_ = ngeoPrintUtils.createPrintMaskPostcompose(
-    getSizeFn, getScaleFn, getRotationFn);
+  this.ogcServers_;
 
   /**
-   * @type {angular.$http.HttpPromise}
+   * @type {Array.<gmfThemes.GmfTheme>}
    * @private
    */
-  this.capabilities_;
+  this.currentThemes_;
+};
 
+
+/**
+ * Init the controller
+ */
+gmf.PrintController.prototype.$onInit = function() {
```

Comment:
you can use classes

Reply:
I think I should do that in another PR (that already a big one and other component still use this notation)